models/CouponModel.php

```diff
@@ -1176,4 +1204,34 @@ public function store_coupon_usage( array $data ) {
 	public function delete_coupon_usage( array $where ) {
 		return QueryHelper::delete( $this->coupon_usage_table, $where );
 	}
+
+	/**
+	 * Get coupon failed error message
+	 *
+	 * @since 3.6.0
+	 *
+	 * @param string $key Error key.
+	 * @param string $variable Variable for placeholder.
+	 *
+	 * @return string
+	 */
+	public function get_coupon_failed_error_msg( string $key, $variable = '' ) {
+		$error_messages = array(
+			'not_found'                 => __( 'Coupon not found', 'tutor' ),
+			'expired'                   => __( 'Coupon expired', 'tutor' ),
+			'invalid'                   => __( 'Coupon invalid', 'tutor' ),
+			'usage_limit_exceeded'      => __( 'Coupon usage limit exceeded', 'tutor' ),
+			'user_usage_limit_exceeded' => __( 'Coupon user usage limit exceeded', 'tutor' ),
+			// translators: $variable.
+			'minimum_purchase'          => sprintf( __( 'This coupon requires a minimum purchase %s', 'tutor' ), $variable ),
+			// translators: $variable.
+			'minimum_quantity'          => sprintf( __( 'This coupon requires minimum purchase of %1$d %2$s', 'tutor' ), $variable, $variable > 1 ? 'quantities' : 'quantity' ),
```

Comment:
Missing placeholder
```
/* translators: %d: variable, %s : single or plural form of quality */
```

Missing translation of `__( 'quantities', 'tutor' ) : __( 'quantity', 'tutor' )`